scripts/generate_version.bash

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Generate version <semantic version of last tag(=release)>+<number of commits from last tagged commit>
```

Comment:
THOUGHT:  using the number of commits from the last tag its a good idea but why not simply using the commit number i.e. short version of it. 

Moreover. Have a look at python's version identification scheme https://peps.python.org/pep-0440/#version-scheme
```
[N!]N(.N)*[{a|b|rc}N][.postN][.devN]
```
Chatty explains it very nicely:

1. `[N!]`: This part is typically used to represent the major version number. The "N" here represents a numeric digit. The "!" is optional and indicates that this part is "bang-prefixed," which suggests that this is an "epoch" version, and it's used to signify a major breaking change in the software that requires special attention.

2. `N(.N)*`: This part represents the minor version number. The "N" is a numeric digit, and the "(.N)*" means that there can be one or more additional digits separated by a dot, indicating sub-minor or patch versions. For example, it can be "2.1," "2.1.3," and so on.

3. `[{a|b|rc}N]`: This part is used to represent pre-release versions. The letters in square brackets indicate pre-release version types:
   - "a" for alpha
   - "b" for beta
   - "rc" for release candidate
   The "N" represents a numeric digit to indicate the specific version of the pre-release. For example, "a3" could represent alpha version 3.

4. `[.postN]`: This part represents a post-release version, typically used for bug fixes or minor updates after the initial release. The "N" here is a numeric digit indicating the specific post-release version.

5. `[.devN]`: This part can be used to indicate development versions or builds. The "N" represents a numeric digit to specify the specific development version or build number.

In summary, this version scheme provides a structured way to represent software versions with major, minor, pre-release, post-release, and development version components. It allows for clear version differentiation and helps users understand the nature of each version (e.g., whether it's a major release, a pre-release version, a post-release update, or a development build).

---

So after reading this, how would be adapted. IMO 
- the release tag could handle `[N!]N(.N)*`
- what about the rest??

Reply:
Here are my thoughts on this: 

1. The reason for using the "distance" to the latest released/tagged commit is that the versions are linearly ordered. So it is easy to say if version A is older/newer than version B. That soothes my OCD and it is quite useful for testing because I can filter out tests by simply adding a `@pytest.skipif(Version(osparc.__version__)) <=Version("0.65.2+4")`. This gets more complicated if I use the commit hash directly. I can see the usefulness of having the commit hash accessible from within the package, but then I would add rather add it in some metadata, I think it is not so useful to have programatically available (whereas I think the version should be).
2. After reading your/chatty's summary I would suggest we something like `0.5..2.dev8`, where the `0.5.2` is the semantic version of the latest release and `dev8` indicates that this client version is 8 commits after the `0.5.2` release.